src/mscorlib/src/System/Threading/LazyInitializer.cs

```diff
@@ -198,11 +197,30 @@ public static T EnsureInitialized<T>(ref T target, ref bool initialized, ref obj
                 return target;
             }
 
-
             return EnsureInitializedCore<T>(ref target, ref initialized, ref syncLock, valueFactory);
         }
 
         /// <summary>
+        /// Initializes a target reference type with a specified function if it has not already been initialized.
+        /// </summary>
+        /// <typeparam name="T">The type of the reference to be initialized. Has to be reference type.</typeparam>
+        /// <param name="target">A reference of type <typeparamref name="T"/> to initialize if it has not already been initialized.</param>
+        /// <param name="syncLock">A reference to an object used as the mutually exclusive lock for initializing
+        /// <paramref name="target"/>. If <paramref name="syncLock"/> is null, a new object will be instantiated.</param>
+        /// <param name="valueFactory">The <see cref="T:System.Func{T}"/> invoked to initialize the reference.</param>
+        /// <returns>The initialized value of type <typeparamref name="T"/>.</returns>
+        public static T EnsureInitialized<T>(ref T target, ref object syncLock, Func<T> valueFactory) where T : class
+        {
+            // Fast path.
+            if (Volatile.Read(ref target) != null)
+            {
+                return target;
+            }
+            
+            return EnsureInitializedCore<T>(ref target, ref syncLock, valueFactory);
+        }
```

Comment:
Nit: I realize you're just copying the style used above; that's fine.  We could subsequently shorten / clean this up as:
```C#
public static T EnsureInitialized<T>(ref T target, ref object syncLock, Func<T> valueFactory) where T : class =>
    Volatile.Read(ref target) ??
    EnsureInitializedCore<T>(ref target, ref syncLock, valueFactory);
```